### PR TITLE
Various tokenizers fixes

### LINKED ISSUE #5393: GPT2Tokenizer.save_pretrained does not work in v3.0.0
# 🐛 Bug

## Information

I am using GPT2Tokenizer and try to save the information about the tokenizer.

In v2.7, I always do

```python
>>> from transformers import GPT2Tokenizer
>>> a = GPT2Tokenizer('./tests/dataloader/dummy_gpt2vocab/vocab.json', './tests/dataloader/dummy_gpt2vocab/merges.txt')
>>> a.save_pretrained("./")
('./vocab.json', './merges.txt', './special_tokens_map.json', './added_tokens.json')
```

And I found it doesn't work when I upgrade to 3.0.
```python
>>> from transformers import GPT2Tokenizer
>>> a = GPT2Tokenizer('./tests/dataloader/dummy_gpt2vocab/vocab.json', './tests/dataloader/dummy_gpt2vocab/merges.txt')
>>> a.save_pretrained("./")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\hzhwcmhf\anaconda3\envs\mix\lib\site-packages\transformers\tokenization_utils_base.py", line 1362, in save_pretrained
    f.write(json.dumps(tokenizer_config, ensure_ascii=False))
  File "C:\Users\hzhwcmhf\anaconda3\envs\mix\lib\json\__init__.py", line 238, in dumps
    **kw).encode(obj)
  File "C:\Users\hzhwcmhf\anaconda3\envs\mix\lib\json\encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "C:\Users\hzhwcmhf\anaconda3\envs\mix\lib\json\encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "C:\Users\hzhwcmhf\anaconda3\envs\mix\lib\json\encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type AddedToken is not JSON serializable
>>> a.save_vocabulary("./")
('./vocab.json', './merges.txt')
```

I know there is ``save_vocabulary`` now, but has ``save_pretrained`` been removed and should not be used?
However, ``save_vocabulary`` does not save special tokens. So how can I save the whole information about the tokenizer.

## Environment info

- `transformers` version: 3.0.0
- Platform: Windows-10-10.0.18362-SP0
- Python version: 3.7.6
- PyTorch version (GPU?): 1.4.0 (True)
- Tensorflow version (GPU?): not installed (NA)
- Using GPU in script?: No
- Using distributed or parallel set-up in script?: No

### LINKED ISSUE #5482: Can't pickle tokenizers ...
Tokenizers are losing all their special tokens when un-pickled.  Not sure if there are other attributes that aren't being rehydrated as well ...

```
hf_tokenizer
# <transformers.tokenization_roberta.RobertaTokenizer at 0x7fc4ce625f10>

hf_tokenizer.special_tokens_map
# {'bos_token': '<s>',
# 'eos_token': '</s>',
# 'unk_token': '<unk>',
# 'sep_token': '</s>',
# 'pad_token': '<pad>',
# 'cls_token': '<s>',
# 'mask_token': '<mask>'}

pickle.dump( hf_tokenizer, open( "save.p", "wb" ) )
hf_tokenizer = pickle.load( open( "save.p", "rb" ) )

hf_tokenizer.special_tokens_map
# {'bos_token': '',
# 'eos_token': '',
# 'unk_token': '',
# 'sep_token': '',
# 'pad_token': '',
# 'cls_token': '',
# 'mask_token': ''}

```

### LINKED ISSUE #5486: Tokenizers throwing warning "The current process just got forked, Disabling parallelism to avoid deadlocks.. To disable this warning, please explicitly set TOKENIZERS_PARALLELISM=(true | false)"
I know this warning is because the transformer library is updated to 3.x.
I know the warning saying to set TOKENIZERS_PARALLELISM = true / false

My question is where should i set TOKENIZERS_PARALLELISM = true  / false
is this when defining tokenizers like 
```
tok = Tokenizer.from_pretrained('xyz', TOKENIZERS_PARALLELISM=True) // this doesn't work
```
or is this when encoding text like
```
tok.encode_plus(text_string, some=some, some=some, TOKENIZERS_PARALLELISM = True) // this also didn't work
```

Suggestions anyone?

### LINKED ISSUE #5490: [ERROR] Tokenizer and TokenizerFast ???
# 🐛 Bug

## Information

Model I am using (Bert, XLNet ...): BERT

Language I am using the model on (English, Chinese ...): 'bert-base-multilingual-cased'

The problem arises when using:
* [ x] the official example scripts: (give details below)
* [ ] my own modified scripts: (give details below)

The tasks I am working on is:
* [ ] an official GLUE/SQUaD task: (give the name)
* [ x] my own task or dataset: (give details below)
## To reproduce

1. `from transformers import *`
2. `tokenizer = BertTokenizerFast.from_pretrained('bert-base-multilingual-cased')`
3. `tokenizer.decode(tokenizer.encode('mở bài lạc trôi'))` --> wrong 

but:

1. `from transformers import *`
2. `tokenizer = BertTokenizer.from_pretrained('bert-base-multilingual-cased')`
3. `tokenizer.decode(tokenizer.encode('mở bài lạc trôi'))` --> true

## Expected behavior
the decode sentence after encoding and decoding using TokenizerFast should be true

## Environment info

     
- `transformers` version:
- Platform: Pytorch and TF
- Python version: 3.6
- PyTorch version (GPU?): GPU
- Tensorflow version (GPU?): 2.2
- Using GPU in script?: Yes
- Using distributed or parallel set-up in script?: No

### PR DESCRIPTION
**Pull request by n1t0**
Fix https://github.com/huggingface/transformers/issues/5486
Fix https://github.com/huggingface/transformers/issues/5482
Fix https://github.com/huggingface/transformers/issues/5393
Fix https://github.com/huggingface/transformers/issues/5490

### CODE REVIEW

File: src/transformers/tokenization_bert.py [hunk 1]
```diff
@@ -606,7 +606,7 @@ def __init__(
         mask_token="[MASK]",
         clean_text=True,
         tokenize_chinese_chars=True,
-        strip_accents=True,
+        strip_accents=None,
```
Comment (n1t0) [inline src/transformers/tokenization_bert.py:609]: The option to strip accents is still available, but not specified by default. If not specified, the behavior is the same than the default implementation.

File: tests/test_tokenization_fast.py [hunk 2]
```diff
@@ -91,8 +91,6 @@ def fast_align_python(self, tokenizer_r, tokenizer_p, tok_case, pretrained_name)
         self.assert_padding(tokenizer_r, tokenizer_p)
         self.assert_create_token_type_ids(tokenizer_r, tokenizer_p)
         self.assert_prepare_for_model(tokenizer_r, tokenizer_p)
-        # TODO: enable for v3.0.0
```
Comment (n1t0) [inline tests/test_tokenization_fast.py:94]: I think we can remove these as the method below doesn't seem to exist.

File: tests/test_tokenization_fast.py [hunk 3]
```diff
@@ -748,29 +746,37 @@ def assert_offsets_with_special_characters(self, tokenizer_r):
             add_special_tokens=True,
         )
 
+        do_lower_case = tokenizer_r.init_kwargs.get("do_lower_case")
```
Comment (n1t0) [inline tests/test_tokenization_fast.py:749]: I don't think we test for both values of `do_lower_case`, but I kept doing it just in case we need it.

**Review by LysandreJik — APPROVED** [review]
Very cool!

File: tests/test_tokenization_fast.py (hunk 3 above)
Comment (thomwolf) [reply tests/test_tokenization_fast.py:749]: Got it, don't hesitate to add tests if you feel like we need some more

**Review by thomwolf — APPROVED** [review]
Awesome!

**Review by LysandreJik — APPROVED** [review]

**Pull request merged**